### DISCUSSION TITLE
Experiment traffic weight

### DISCUSSION THREAD

**Discussion by derjust** [argoproj/argo-rollouts#1828] [Q&A]
### Background 
We usually do our wargames by having a dedicated service instance and push more traffic to it than all the other instances.
That way we can simulate how services will behave under higher load in general & see _how_ they will fail.
We use this approach as our production traffic is very hard to artificially generate.

### Question
How does the traffic distribution work when there is an experiment started in a rollout-scenario?

We use Argo Rollouts very successful with Traffic Management against AWS ALB & Istio.
How do experiments work?
Are the pods started and they get the same traffic percentage wise as regular pods? 

Assuming 9 regular pods, every gets 1/9 of the traffic.
Adding 1 pod with an experiment, is now each pod getting 1/10 of the traffic?
Can we somehow force the experiment to take more of the traffic, preferably very manually via `kubectl` or something? (Also with some kind of `kubectl experiment abort` that just gets us back to the pre-experiment-running state)

**Comment by harikrongali** [comment]
as experiment creates POD and work in similar way as in canary. you can set weight for the experiment . If 20% is set for the experiment and this will be deducted from total traffic. You can time the experiment and after specified duration, it will return to pre experiment state. Yes abort does the same as well
https://argoproj.github.io/argo-rollouts/features/experiment/

**Reply by derjust** [reply]
Thank you @harikrongali - that sounds promising!
>If 20% is set for the experiment

this is probably the part that I am missing out: I do not see the traffic weights being applied via an experiment - i only see it when the experiment is used as part of a rollout.
I see `weight` on `experiments.argoproj.io` only as part of the `node`/`pod`[`anti`]-`affinity`

Is there a `spec` document somewhere similar to the [`rollout`](https://argoproj.github.io/argo-rollouts/features/specification/) that lists all possible parameters?

**Comment by harikrongali** [comment]
https://argoproj.github.io/argo-rollouts/features/experiment/#integration-with-rollouts
https://argoproj.github.io/argo-rollouts/features/experiment/#experiment-spec
